test/dirent_lookup.cpp

```diff
@@ -105,42 +105,56 @@ class DirentLookupTest : public :: testing::Test
 
 TEST_F(DirentLookupTest, ExactMatch)
 {
+  zim::DirentLookup<GetDirentMock> direntLookup(&dirents);
+  zim::FastDirentLookup<GetDirentMock> fast_direntLookup(&dirents, 4);
+
 #define CHECK_EXACT_MATCH(expr, expected_value) \
   { \
-    const auto result = expr; \
-    ASSERT_TRUE(result.first); \
-    ASSERT_EQ(result.second.v, expected_value); \
+    const auto direntLookupFindResult = expr; \
+    ASSERT_TRUE(direntLookupFindResult.first); \
+    ASSERT_EQ(direntLookupFindResult.second.v, expected_value); \
+  } \
+  { \
+    const auto fast_direntLookupFindResult = fast_##expr; \
```

Comment:
`CHECK_EXACT_MATCH('A', "aa", 0);` is less readable since it conceals the operation being tested and you have to look into the macro definition.

But I slightly improved my version, by removing some code duplication and making the macro definitions more concise and less cluttered and thus easier to understand.

Reply:
This is better yes.